File: Lab5/C#/Printer.cs
using System;

namespace Lab5_InheritanceAndPolymorphism
{
    public class Printer
    {
        public static void PrintSystem(TSystemLinearEquation system)
        {
            Console.WriteLine($"System: \n{system}");
        }

        public static void PrintRoots(double[] roots)
        {
            if (roots != null)
            {
                for (int i = 0; i < roots.Length; i++)
                    Console.WriteLine($"{(char) (i + 120)} = {roots[i]}");
                Console.WriteLine("\n\n");
            }
            else
                Console.WriteLine("System has infinite quantity of roots.\n\n");
        }
    }
}

File: Lab5/Python/Printer.py
class SystemsPrinter:
    @staticmethod
    def print_system(lin_system):
        print('System:')
        print(f'{lin_system}')

    @staticmethod
    def print_roots(roots):
        if roots is not None:
            i = 0
            for root in roots:
                print(f'{chr(i + 120)} = {round(root, 3)}')
                i += 1
            print('\n')
        else:
            print('Equation has infinite roots')
